### PR TITLE
Show boostdep data in library/release detail and report.

### LINKED ISSUE #1336: per-release library boostdep reports
Each library should track its own boostdep report per-release. The UI should show when dependencies are added or removed relative to the previous release.

### PR DESCRIPTION
**Pull request by brianjp93**
- fixes #1336
- If a libraryversion does not have any known dependencies in the version in question or the prior version, stats are not shown.  There is the case where if a library goes from 0 to n dependencies or vice-versa, that stat will not be shown.  If this is a necessary feature, more work will need to be done to facilitate it.  We are only keeping a many-to-many relationship of LibraryVersions to Libraries so we cannot distinguish between if there are no dependencies or if we were not able to fetch them through the boostdep github action.

### CODE REVIEW

**Comment by brianjp93** [comment]
# library detail page
- I chose green for removed dependencies and red for added.
- The green is a bit light, I'll leave it for someone else to choose the exact color you want, if any.
image
image

# release page
image

# release report
image

**Comment by GregKaleka** [comment]
I think I would have made the opposite color choice (green for added, red for removed). I'll let others chime in on specific shades.

**Review by GregKaleka — APPROVED** [review]
Code LGTM. Hold off on merging until we get UI feedback from Rob and/or Vinnie.

File: templates/admin/release_report_detail.html
Comment (GregKaleka) [inline templates/admin/release_report_detail.html:353]: IMO the arrangement you used for dependencies is preferred.

```suggestion
                  There {{ item.issues.opened|pluralize:"was,were" }} {{ item.issues.opened }} issue{{ item.issues.opened|pluralize }} opened
                  and {{ item.issues.closed }} issue{{ item.issues.closed|pluralize }} closed
```

**Comment by rbbeeston** [comment]
I understand why you did the color choices that way, that's fine, it expresses what's needed. I think it would also be helpful to put a "(*New)" in smaller text next to the word "Dependencies" and put a preceding asterisk next to each new dependency for color blind users and for more clarity.

**Comment by brianjp93** [comment]
image

**Comment by rbbeeston** [comment]
Can we have the added/removed fonts smaller? like an h4 instead of h3

**Comment by brianjp93** [comment]
image

**Review by rbbeeston — APPROVED** [review]
I think this looks good and can be merged now.

**Pull request merged**
